### ISSUE TITLE
Misc: Update enzyme to have React Hooks support

### ISSUE THREAD

**Issue by gbonaspetti** [tracim/tracim#5223]
## Description

Initially the Popover component had been made in React Hook, but the frontend unit tests failed systematically with error `Invalid hook call. Hooks can only be called inside of the body of a function component`. Even if, when the code was run in local, there were no errors in the console.

**Failing tests**: 

- frontend/test/src/container/AdminAccount.spec.js  
- frontend/test/src/container/Account.spec.js
- frontend/test/src/container/NotificationWall.spec.js

**Component architecture**: https://github.com/tracim/tracim/blob/02d661186b625bdf114e78fc6b3de523a0e89399/frontend_lib/src/component/Popover/Popover.jsx

After investigating a little further, I discovered that the enzyme library that is used in the unit tests does not have full hook support yet (https://www.npmjs.com/package/enzyme#react-hooks-support).

In the related open issue there are many people who have given workaround alternatives and we can follow the progress of the support (which for now is higher in tests that use shallow than mount): https://github.com/enzymejs/enzyme/issues/2011

For time reasons I only made the component into a class, but for the future it would be important to **update the enzyme version**. Even because, theoretically, the invariant violation problem using useState has already been solved in recent versions: https://github.com/enzymejs/enzyme/issues/1938

## todo
- [x] **Make react external in frontend_lib**
  - This is the most important change that fixes the unit tests using component using hooks from frontend_lib
- [x] Use gbonaspetti's Popover.jsx hooks implementation
- [x] Update `frontend/test/src/container/NotificationWall.spec.js` comment to link my last comment for advises
- [x] Upgrade enzyme to latest (3.10.0 => 3.11.0)
- [x] Rationalize usage of reactstrapPopoverHack in Account.spec.js
- [x] add react as peer dep in frontend lib
- [x] check that everything works fine with adding react as external in frontend_lib
- [x] Check at runtime that popover still works

## PR LInk
https://github.com/tracim/tracim/pull/5577

**Comment by Skylsmoi** [comment]
I'm linking this issue with https://github.com/tracim/tracim/issues/3816
because both are about upgrading js libraries

**Comment by Skylsmoi** [comment]
I ran some tests.

Unit tests are working.

I ran unit tests with gbonaspetti's hook implementation for Popover.jsx and unit tests still runs fine.

It looks like there is no need for any library update regarding unit tests and hooks.

___

This makes the comment in file frontend/test/src/container/NotificationWall.spec.js wrong.
Still, I tried to uncomment the file and run the tests and none of them worked. But the isn't related to enzyme, it's because the tests expect to get the instance of the wrapper but it always return null. When it was a class component, the tests called the functions of the component from its instance, now it's a functional component, this can't be done.
After looking at some of these functions, it seems they have to reason to be part of the component since they are immutable.

I would advise to put every immutable function (the ones that don't call any redux or local state change) in a notification helper file to make them more easilly unit testable.
